### ISSUE TITLE
Update python3 references to vpython3

### ISSUE THREAD

**Issue by a-wallen** [flutter/flutter#108474]
### TL;DR
Update `python3` references with `vpython3` for consistent, reproducible builds. 

### Root cause
Instructions for [setting up the Engine development environment](https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment) guide users to install **both** Chromium's [depot_tools](https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up) and Python.
> - Chromium's [depot_tools](https://commondatastorage.googleapis.com/chrome-infra-docs/flat/depot_tools/docs/html/depot_tools_tutorial.html#_setting_up) (make sure it's in your path). We use the gclient tool from depot_tools.
Make sure to install gclient by running ./gclient in the command line.
> - Python (used by many of our tools, including gclient).

Hence, developers _correctly_ assume that they may rely on `python3` when developing scripts, but a user installed `python3` can lead to unexpected behavior as seen in flutter/flutter#108122 and [confusion](https://github.com/flutter/flutter/issues/108122#issuecomment-1195322469).

The history shows that we intended to use `vpython3` for consistent, reproducible builds, but due the following upstream issue, flutter/flutter#88719, those references were reverted back to `python3`. 

WIP Todo List:
1. Update `python3` references to `vpython3`
- https://github.com/flutter/engine/blob/main/tools/gn.bat
- https://github.com/flutter/engine/blob/main/tools/gn#L1
- https://github.com/flutter/buildroot/blob/master/.gn
2. Update [Setting up the Engine development environment](https://github.com/flutter/flutter/wiki/Setting-up-the-Engine-development-environment)

**Comment by a-wallen** [comment]
https://github.com/flutter/buildroot/blob/master/.gn resolved by flutter/buildroot#590 and flutter/engine

**Comment by loic-sharma** [comment]
To catch unexpected uses of python3, consider putting a python3 script at the front of your path that fails loudly. You'll want to make sure the dev workflow still works as expected, including build, git push (for git hooks), etc...
